timemachine/fe/free_energy.py

```diff
@@ -1449,6 +1450,7 @@ def get_flattened_params(initial_state: InitialState) -> NDArray:
 
     samples_by_state: list[Trajectory] = [Trajectory.empty() for _ in initial_states]
     replica_idx_by_state_by_iter: list[list[ReplicaIdx]] = []
+    water_sampler_proposals: list[list[tuple[int, int]]] = []
```

Comment:
nit/naming: following the pattern of the other fields, can we indicate the dimensions in the name? (e.g. maybe `water_sampler_proposals_by_state_by_iter`, assuming that is correct?